internal/pkg/cli/env_delete.go

```diff
@@ -5,29 +5,129 @@ package cli
 
 import (
 	"errors"
+	"fmt"
+	"strings"
 
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	termprogress "github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/progress"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
 	"github.com/spf13/cobra"
 )
 
+const (
+	fmtDeleteEnvPrompt = "Delete environment %s from project %s?"
+)
+
+const (
+	fmtDeleteEnvStart    = "Deleting environment %s from project %s."
+	fmtDeleteEnvFailed   = "Failed to delete environment %s from project %s: %v."
+	fmtDeleteEnvComplete = "Deleted environment %s from project %s."
+)
+
 // DeleteEnvOpts holds the fields needed to delete an environment.
 type DeleteEnvOpts struct {
-	Env              string
+	// Arguments and flags.
+	EnvName          string
 	SkipConfirmation bool
+
+	// Interfaces for dependencies.
+	storeClient  archer.EnvironmentStore
+	rgClient     resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI
+	iamClient    iamiface.IAMAPI
+	deployClient environmentDeployer
+	prog         progress
+
+	// Cached objects to avoid multiple requests.
+	env *archer.Environment
+
+	*GlobalOpts
 }
 
 // Ask is a no-op for this command.
 func (opts *DeleteEnvOpts) Ask() error {
 	return nil
 }
 
-// Validate returns an error if the environment name does not exist in the project, or there are existing applications
-// in the environment, or the environment is not available using the EnvironmentManagerRole.
+// Validate returns an error if the environment name does not exist in the project, or if there are live applications under the environment.
 func (opts *DeleteEnvOpts) Validate() error {
+	// Validate that the environment is stored in SSM.
+	env, err := opts.storeClient.GetEnvironment(opts.ProjectName(), opts.EnvName)
+	if err != nil {
+		return fmt.Errorf("get environment %s metadata in project %s: %w", opts.EnvName, opts.ProjectName(), err)
+	}
+	opts.env = env
+
+	if opts.rgClient == nil {
+		// Tests mock the client.
+		if err := opts.initClientsFromRole(opts.env.ManagerRoleARN, opts.env.Region); err != nil {
+			return err
+		}
+	}
+
+	// Look up applications by searching for cloudformation stacks in the environment's region.
+	stacks, err := opts.rgClient.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
```

Comment:
TODO - put this somewhere _else_. Maybe describe.

Reply:
Added TODO